src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -608,7 +627,7 @@ private PackageRestoreInputs DetermineRestoreInputs()
         /// <returns>The full path of the solution file. Or null if no solution file can be found.</returns>
         private string GetSolutionFile(string solutionFileOrDirectory)
         {
-            if (File.Exists(solutionFileOrDirectory))
+            if (File.Exists(solutionFileOrDirectory) && IsSolutionOrProjectFile(solutionFileOrDirectory))
             {
                 return Path.GetFullPath(solutionFileOrDirectory);
```

Comment:
From the name here it looks like this is suppose to find the solution file from a directory, but using the above check it would not make it here on a dir.